src/modules/nodejs.rs

```diff
@@ -56,19 +58,26 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
             })
             .map(|variable| match variable {
                 "version" => {
-                    let version = nodejs_version
+                    let node_ver = nodejs_version
                         .deref()
                         .as_ref()?
                         .trim_start_matches('v')
                         .trim();
 
                     VersionFormatter::format_module_version(
                         module.get_name(),
-                        version,
+                        node_ver,
                         config.version_format,
                     )
                     .map(Ok)
                 }
+                "engines_version" => {
+                    let node_ver = nodejs_version.as_deref();
+                    let eng_ver = get_engines_version(context)?;
```

Comment:
The function `get_engines_version()` reads a file and parses a JSON. Both these operations are expensive. Call it only once, store the value and use it any number of times it is needed.
Also, `check_engines_version()` should be called only once per execution.

Reply:
Compared to stuff like getting the software version, it should be relatively cheap, but I agree that it could be a worthwhile optimization to store the result with a `Lazy<_>` variable.

Reply:
I can try these both, gimme a sec